src/EntangledUnits/EntangledUnits.jl

```diff
@@ -254,6 +258,13 @@ function entangle_system(sys::System{M}, units) where M
     spin_infos = [i => Moment(s=(N-1)/2, g=1.0) for (i, N) in enumerate(Ns_contracted)]  # TODO: Decisions about g-factor 
     sys_entangled = System(contracted_crystal, spin_infos, :SUN; dims)
 
+    # Transfer rng from origin system to entangled system
+    sys_entangled.rng.s0 = sys.rng.s0
+    sys_entangled.rng.s1 = sys.rng.s1
+    sys_entangled.rng.s2 = sys.rng.s2
+    sys_entangled.rng.s3 = sys.rng.s3
+    sys_entangled.rng.s4 = sys.rng.s4
```

Comment:
Can this be replaced with `sys_entangled.rng = copy(sys.rng)`? The fields of an `rng` are private.

Reply:
It's a workaround for the fact that the `rng` field of a `System` is `const` -- admittedly a very ugly and brittle workaround. I can think of other ways of doing it, but they're all ugly in similar ways.

I'll give it some thought.

Reply:
Per your suggestion, used `Base.copy!` instead of accessing private fields.